### ISSUE TITLE
Question: On how to run LMNN metric-learn tool?

### ISSUE THREAD

**Issue by ivan-marroquin** [all-umass/metric-learn#75]
Hi,
I read the documentation and also checked the example, but there is still an step not clear to me on how to proceed. 

In my case, I want to validate the cluster centers against a training data. So, I have the following:
- labels= [1, 2, 3] -> Labels
- cluster centers an array of size 3 x 4 -> Cluster_Centers
- a training data of size 801 x 4 -> Train_Data

I understood well these steps:
- build the LMNN model by
lmnn= LMNN(k= 1, learn_rate= 1e-7) # I use k=1 given the few samples that I have for the cluster centers

- fit the model using the cluster centers and labels:
lmnn.fit(Clusters_Centers, Labels)

- project the training data onto the learned space
proj_train_data= lmnn.transform(Train_Data)

Now, it is not clear to me on how to proceed afterwards. I want to be able to measure a distance between the "proj_train_data" with respect to each of the cluster centers. So then, I can assign a label to the result based on the minimum distance to a given cluster center. How should I proceed?

Many thanks,
Ivan

**Comment by perimosocordiae** [comment]
To compare your cluster centers with the projected training set, you'll want to project both:

``python
proj_centers = lmnn.fit_transform(Clusters_Centers, Labels)
proj_train_data = lmnn.transform(Train_Data)
```

I'm not sure that training LMNN with only the centers of each cluster is going to be very effective, though.

**Comment by ivan-marroquin** [comment]
Thanks for the prompt answer! 

I noticed that I forgot to mention that my Train_Data also contains labels. So, I followed your advice:

lmnn= LMNN(k= 1, learn_rate= 1e-7)
lmnn.fit(Train_Data, Labels_Train_Data)
proj_centers= lmnn.transform(Clusters_Centers)
proj_train_data= lmnn.transform(Train_Data)

Then, I used the Euclidean distance to compare both projections.

By the way, the command that you mentioned in your answers gives the following error message:

proj_well_data= lmnn.fit_transform(Clusters_Centers, Labels)

AttributeError: 'python_LMNN' object has no attribute 'fit_transform'
